include/zephyr/drivers/bluetooth.h

```diff
@@ -89,8 +89,18 @@ enum bt_hci_bus {
 #define BT_DT_HCI_NAME_GET(node_id) DT_PROP_OR(node_id, bt_hci_name, "HCI")
 #define BT_DT_HCI_NAME_INST_GET(inst) BT_DT_HCI_NAME_GET(DT_DRV_INST(inst))
 
+/*
+ * When HCI bus is IPM DT_ENUM_IDX returns the value 12.
+ * This value doesn'r exist in bt_hci_bus enumeration.
+ * Use COND_CODE_1 to handle this case by mapping value
+ * 12 to BT_HCI_BUS_IPC.
+ */
 #define BT_DT_HCI_BUS_GET(node_id) \
-	UTIL_CAT(BT_HCI_BUS_, DT_STRING_UPPER_TOKEN_OR(node_id, bt_hci_bus, VIRTUAL))
```

Comment:
Did you try the `IDENTITY()` option? If that works, it'd be simpler than this fix. I.e. `IDENTITY(DT_STRING_UPPER_TOKEN_OR(node_id, bt_hci_bus, VIRTUAL))`

Reply:
[This Stackoverflow post](https://stackoverflow.com/questions/45404912/c-is-it-possible-to-expand-marcos-before-pasting-but-not-recursively) says it's not possible. There is no way to expand a macro like `DT_STRING_UPPER_TOKEN_OR` without causing its result to be expanded too.